fri/src/two_adic_pcs.rs

```diff
@@ -255,45 +269,56 @@ where
                 (mats, points)
             })
             .collect_vec();
-        let mats = mats_and_points
-            .iter()
-            .flat_map(|(mats, _)| mats)
-            .collect_vec();
 
-        let global_max_height = mats.iter().map(|m| m.height()).max().unwrap();
+        // Find the maximum height of a matrix in the batch. TODO: This should be given by the matrices in the first round right?
```

Comment:
I don't believe that's always the case, i.e. the "round 0" might be preprocessed table commitments, and there might be a longer trace in "round 1" (the main traces) if it's from an AIR that doesn't use preprocessing.